src/NServiceBus.Transport.SQS.AcceptanceTests/NativeIntegration/When_receiving_a_native_message.cs

```diff
@@ -0,0 +1,203 @@
+﻿namespace NServiceBus.AcceptanceTests.NativeIntegration
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Text;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using System.Xml.Linq;
+    using AcceptanceTesting;
+    using AcceptanceTesting.Customization;
+    using Amazon.S3.Model;
+    using Amazon.SQS.Model;
+    using Configuration.AdvancedExtensibility;
+    using EndpointTemplates;
+    using NUnit.Framework;
+    using Recoverability;
+    using Settings;
+    using Transport.SQS;
+
+    public class When_receiving_a_native_message : NServiceBusAcceptanceTest
+    {
+        static readonly string MessageToSend = new XDocument(new XElement("Message", new XElement("ThisIsTheMessage", "Hello!"))).ToString();
+
+        [Test]
+        public async Task Should_be_processed_when_messagetypefullname_present()
+        {
+            var context = await Scenario.Define<Context>()
+                .WithEndpoint<Receiver>(c => c.When(async _ =>
+                {
+                    await SendNativeMessage(new Dictionary<string, MessageAttributeValue>
+                    {
+                        {"MessageTypeFullName", new MessageAttributeValue {DataType = "String", StringValue = typeof(Message).FullName}}
+                    });
+                }))
+                .Done(c => c.MessageReceived != null)
+                .Run();
+
+            Assert.AreEqual("Hello!", context.MessageReceived);
+        }
+
+        [Test]
+        public async Task Should_fail_when_messagetypefullname_not_present()
+        {
+            var cancellationTokenSource = new CancellationTokenSource();
+            try
+            {
+                await Scenario.Define<Context>()
+                    .WithEndpoint<Receiver>(c =>
+                    {
+                        c.CustomConfig((cfg, ctx) =>
+                        {
+                            ctx.ErrorQueueAddress = cfg.GetSettings().ErrorQueueAddress();
+                        });
+                        c.When(async (session, ctx) =>
+                        {
+                            await SendNativeMessage(new Dictionary<string, MessageAttributeValue>
+                            {
+                                // unfortunately only the message id attribute is preserved when moving to the poison queue
```

Comment:
Move to error queue does the following

```
                await sqsClient.SendMessageAsync(new SendMessageRequest
                {
                    QueueUrl = errorQueueUrl,
                    MessageBody = message.Body,
                    MessageAttributes =
                    {
                        [Headers.MessageId] = new MessageAttributeValue
                        {
                            StringValue = messageId,
                            DataType = "String"
                        }
                    }
                }, CancellationToken.None).ConfigureAwait(false);
                // The MessageAttributes on message are read-only attributes provided by SQS
                // and can't be re-sent. Unfortunately all the SQS metadata
                // such as SentTimestamp is reset with this send.
```

I have a hunch the comment is not correct and we can actually resent the attributes easily by doing this

```
                MessageAttributes = message.MessageAttributes.ToDictionary(x => x.Key, x => new SnsMessageAttributeValue
                {
                    DataType = x.Value.DataType,
                    StringValue = x.Value.StringValue,
                    BinaryValue = x.Value.BinaryValue,
                })
```

because right now moving to poison queue looses all attributes except the message id. Let's discuss